### ISSUE TITLE
OpenSHMEM-on-UCX: wireup.c assert failed fatal message

### ISSUE THREAD

**Issue by tonycurtis** [openucx/ucx#1998]
Since a commit last Thursday or Friday (git HEAD, master) I am getting the following kind of error message from UCX.  I assumed at first I had broken something but it also happens with OSHMEM in OpenMPI, e.g. 2.1.2 and git-head-master as of a few minutes ago with UCX.

This is working OK with UCX release 1.2.1 and up to and incl. e.g. commit e171fc1e2aa4ccb21ac71f74e09b7c8b2b0e621c.

```
#include <shmem.h>
int main()
{
    shmem_init();
    shmem_finalize();
    return 0;
}
```

compiling/running with e.g. OpenMPI/OSHMEM on local mlx5 cluster gives this (and gdb backtrace appended)

```
[cn004:16556:0]      wireup.c:309  Assertion `ep->flags & UCP_EP_FLAG_CONNECT_REP_SENT' failed
==== backtrace ====
    0  /gpfs/home/arcurtis/opt/ucx/git/lib/libucs.so.0(__ucs_abort+0x111) [0x7fffe9ca85a1]
    1  /gpfs/home/arcurtis/opt/ucx/git/lib/libucp.so.0(+0x30fc9) [0x7fffea574fc9]
    2  /gpfs/home/arcurtis/opt/ucx/git/lib/libucp.so.0(+0x32619) [0x7fffea576619]
    3  /gpfs/home/arcurtis/opt/ucx/git/lib/libuct.so.0(+0x29b69) [0x7fffea30ab69]
    4  /gpfs/home/arcurtis/opt/ucx/git/lib/libucp.so.0(+0x12bb2) [0x7fffea556bb2]
    5  /gpfs/home/arcurtis/opt/ucx/git/lib/libucp.so.0(ucp_worker_progress+0x27) [0x7fffea557d97]
    6  /gpfs/home/arcurtis/opt/openmpi/gnu/git/lib/openmpi/mca_spml_ucx.so(spml_ucx_progress+0x17) [0x7fffe2da62a7]
    7  /gpfs/home/arcurtis/opt/openmpi/gnu/git/lib/libopen-pal.so.0(opal_progress+0x40) [0x7ffff74dd2c0]
    8  /gpfs/home/arcurtis/opt/openmpi/gnu/git/lib/libmpi.so.0(ompi_comm_nextcid+0x145) [0x7ffff7aa56b5]
    9  /gpfs/home/arcurtis/opt/openmpi/gnu/git/lib/libmpi.so.0(ompi_comm_create_group+0xc4) [0x7ffff7aa0d34]
   10  /gpfs/home/arcurtis/opt/openmpi/gnu/git/lib/openmpi/mca_scoll_mpi.so(mca_scoll_mpi_comm_query+0x1ca) [0x7fffe299885a]
   11  /gpfs/home/arcurtis/opt/openmpi/gnu/git/lib/liboshmem.so.0(mca_scoll_base_select+0x662) [0x7ffff7dd5be2]
   12  /gpfs/home/arcurtis/opt/openmpi/gnu/git/lib/liboshmem.so.0(mca_scoll_enable+0x21) [0x7ffff7dd4c71]
   13  /gpfs/home/arcurtis/opt/openmpi/gnu/git/lib/liboshmem.so.0(oshmem_shmem_init+0x328) [0x7ffff7d9d718]
   14  /gpfs/home/arcurtis/opt/openmpi/gnu/git/lib/liboshmem.so.0(shmem_init+0x28) [0x7ffff7da0258]
   15  ./a.out() [0x40065d]
   16  /lib64/libc.so.6(__libc_start_main+0xfd) [0x3b84a1ed5d]
   17  ./a.out() [0x400599]
===================
[cn004:16556:0] Process frozen..
```

```
#0  0x0000003b84e0effd in pause () from /lib64/libpthread.so.0
#1  0x00007fffe9ca6db5 in ucs_debug_freeze ()
    at ../../../ucx-git/src/ucs/debug/debug.c:709
#2  0x00007fffe9ca7744 in ucs_error_freeze (
    error_type=0x7fffea57a853 "assertion failure",
    message=0x7fffffffbcb0 "     wireup.c:309  Assertion `ep->flags & UCP_EP_FLAG_CONNECT_REP_SENT' failed") at ../../../ucx-git/src/ucs/debug/debug.c:828
#3  0x00007fffe9ca791b in ucs_handle_error (
    error_type=0x7fffea57a853 "assertion failure",
    message=<value optimized out>)
    at ../../../ucx-git/src/ucs/debug/debug.c:992
#4  0x00007fffe9ca85a1 in __ucs_abort (
    error_type=0x7fffea57a853 "assertion failure",
    file=0x7fffea580ad0 "../../../ucx-git/src/ucp/wireup/wireup.c", line=309,
    function=<value optimized out>, message=<value optimized out>)
    at ../../../ucx-git/src/ucs/debug/log.c:228
#5  0x00007fffea574fc9 in ucp_wireup_process_ack (
    worker=<value optimized out>, uuid=<value optimized out>)
    at ../../../ucx-git/src/ucp/wireup/wireup.c:309
#6  0x00007fffea576619 in ucp_wireup_msg_handler (arg=0x7fffead22010,
    data=0x7fffe1170cc1, length=<value optimized out>,
    flags=<value optimized out>)
    at ../../../ucx-git/src/ucp/wireup/wireup.c:338
#7  0x00007fffea30ab69 in uct_iface_invoke_am (arg=0xabb010)
    at /gpfs/home/arcurtis/src/ucx/build/../ucx-git/src/uct/base/uct_iface.h:514
#8  uct_ib_iface_invoke_am_desc (arg=0xabb010)
    at /gpfs/home/arcurtis/src/ucx/build/../ucx-git/src/uct/ib/base/ib_iface.h:206
#9  uct_rc_verbs_iface_handle_am (arg=0xabb010)
    at ../../../ucx-git/src/uct/ib/rc/verbs/rc_verbs_common.h:209
#10 uct_rc_verbs_iface_poll_rx_common (arg=0xabb010)
    at ../../../ucx-git/src/uct/ib/rc/verbs/rc_verbs_common.h:232
#11 uct_rc_verbs_iface_progress (arg=0xabb010)
    at ../../../ucx-git/src/uct/ib/rc/verbs/rc_verbs_iface.c:121
#12 0x00007fffea556bb2 in ucs_callbackq_dispatch (cbq=<value optimized out>)
    at /gpfs/home/arcurtis/src/ucx/build/../ucx-git/src/ucs/datastruct/callbackq.h:166
#13 0x00007fffea557d97 in uct_worker_progress (worker=0x7fffead22010)
    at /gpfs/home/arcurtis/src/ucx/build/../ucx-git/src/uct/api/uct.h:1623
#14 ucp_worker_progress (worker=0x7fffead22010)
    at ../../../ucx-git/src/ucp/core/ucp_worker.c:1209
#15 0x00007fffe2da62a7 in spml_ucx_progress ()
   from /gpfs/home/arcurtis/opt/openmpi/gnu/git/lib/openmpi/mca_spml_ucx.so
#16 0x00007ffff74dd2c0 in opal_progress ()
   from /gpfs/home/arcurtis/opt/openmpi/gnu/git/lib/libopen-pal.so.0
#17 0x00007ffff7aa56b5 in ompi_comm_nextcid ()
---Type <return> to continue, or q <return> to quit---
   from /gpfs/home/arcurtis/opt/openmpi/gnu/git/lib/libmpi.so.0
#18 0x00007ffff7aa0d34 in ompi_comm_create_group ()
   from /gpfs/home/arcurtis/opt/openmpi/gnu/git/lib/libmpi.so.0
#19 0x00007fffe299885a in mca_scoll_mpi_comm_query ()
   from /gpfs/home/arcurtis/opt/openmpi/gnu/git/lib/openmpi/mca_scoll_mpi.so
#20 0x00007ffff7dd5be2 in mca_scoll_base_select ()
   from /gpfs/home/arcurtis/opt/openmpi/gnu/git/lib/liboshmem.so.0
#21 0x00007ffff7dd4c71 in mca_scoll_enable ()
   from /gpfs/home/arcurtis/opt/openmpi/gnu/git/lib/liboshmem.so.0
#22 0x00007ffff7d9d718 in oshmem_shmem_init ()
   from /gpfs/home/arcurtis/opt/openmpi/gnu/git/lib/liboshmem.so.0
#23 0x00007ffff7da0258 in shmem_init ()
   from /gpfs/home/arcurtis/opt/openmpi/gnu/git/lib/liboshmem.so.0
#24 0x000000000040065d in main ()
```

**Comment by yosefe** [comment]
fixed by #2032

**Comment by tonycurtis** [comment]
+1
